### PR TITLE
feat(node): v12.12

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#12.12.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Repeat of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39250
Also cleaned up some incorrectly declared exports.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for non-npm package Node.js 12.11
+// Type definitions for non-npm package Node.js 12.12
```
Comment (galkin) [inline types/node/index.d.ts]: should we use 12.13 here?

Comment (SimonSchick) [reply types/node/index.d.ts]: 12.13 didn't really introduce anything new.

Comment (galkin) [reply types/node/index.d.ts]: So, let's use 12.13 here. Could we?

**Review by Flarna — APPROVED** [review]

File: types/node/index.d.ts (hunk 1 above)
Comment (SimonSchick) [reply types/node/index.d.ts]: There is no advantage in doing so and would just trigger another build and delay things further, I will leave as is *unless* another change will be made.

**Review by trivikr — APPROVED** [review]

**Comment by ddehghan** [comment]
Hi guys - we just started getting this error in our CI:

 ../node_modules/@grpc/grpc-js/build/src/metadata.d.ts(71,29): error TS2694: Namespace '"http2"' has no exported member 'OutgoingHttpHeaders'.

seems that it is a regression.

**Comment by Flarna** [comment]
@ddehghan PR to fix this: #39992

**Pull request merged**